doc/reference/fortran.rst

```diff
@@ -118,3 +118,180 @@ The basic error handler is an allocatable derived type, available from ``mctc_en
    end
 
 An unhandled error might get dropped by the next procedure call.
+
+
+Performing calculations
+-----------------------
+
+An example for performing a calculation with DFT-D4 is shown below
+
+.. code-block:: fortran
+
+   subroutine calc_dftd4(error, mol, method, energy, model, gradient, sigma)
+      use mctc_env, only : wp, error_type, fatal_error
+      use mctc_io, only : structure_type
+      use dftd4, only : damping_param, get_rational_damping, get_dispersion
+      use dftd4_cutoff, only : realspace_cutoff
+      use dftd4_model, only : dispersion_model
+      use dftd4_model_d4, only : d4_model, new_d4_model
+      use dftd4_model_d4s, only : d4s_model, new_d4s_model
+      use dftd4_utils, only : lowercase
+      implicit none
+
+      !> Error handling
+      type(error_type), allocatable, intent(out) :: error
+
+      !> Molecular structure data
+      type(structure_type), intent(in) :: mol
+      
+      !> Method name for which parameters should be used
+      character(len=*), intent(in) :: method
+      
+      !> Dispersion energy
+      real(wp), intent(out) :: energy
+
+      !> Dispersion model (D4 or D4S)
+      character(len=*), intent(in), optional :: model
+
+      !> Dispersion gradient
+      real(wp), intent(out), contiguous, optional :: gradient(:, :)
+      
+      !> Dispersion virial
+      real(wp), intent(out), contiguous, optional :: sigma(:, :)
+
+      class(damping_param), allocatable :: param
+      class(dispersion_model), allocatable :: disp
+
+      ! Get D4 damping parameters for given method
+      call get_rational_damping(method, param, s9=1.0_wp)
+      if (.not.allocated(param)) then
+         call fatal_error(error, "No parameters for '"//method//"' available")
+         return
+      end if
+
+      ! Initialize D4/D4S model
+      call new_dispersion_model(error, d4, mol)
+      if (allocated(error)) return
+      
+      call get_dispersion(mol, disp, param, realspace_cutoff(), energy, &
+         & gradient, sigma)
+
+   end subroutine calc_dftd4
+
+
+Complete Example
+----------------
+
+A minimal program using the snippets from above could look like this:
+
+.. code-block:: fortran
```

Comment:
If this should be used with fpm, then we should be explicit about in what file we will save it. I think `:caption: app/main.f90` is appropriate here.